### ISSUE TITLE
Setting member default values causes false detection of a dependency loop

### ISSUE THREAD

**Issue by papparapa** [ziglang/zig#17533]
### Zig Version

0.12.0-dev.926+3be8490d8

### Steps to Reproduce and Observed Behavior

The following codes cannot be compiled.

```zig
const A = struct {
    b: ?*B = null,
    n: usize,
};

const B = struct {
    a: A = A{ .n = 1 },
};

pub fn main() void {
    _ = A{ .n = 10 };
}
```

```bash
$ zig build-exe false_dependency_loop.zig
false_dependency_loop.zig:1:11: error: struct 'false_dependency_loop.A' depends on itself
const A = struct {
          ^~~~~~
$
```

### Expected Behavior

I expect the compile gets successful since A.b is a pointer.

**Comment by kcbanner** [comment]
https://github.com/ziglang/zig/pull/17490 may solve this

**Comment by kcbanner** [comment]
Confirming that PR does fix this case.

**Comment by papparapa** [comment]
I found that this issue was resolved in 0.12.0-dev.1504+81219586b at the latest, meybe thanks to #17692
